packages/server/lib/controllers/v1/environment/otlp/postSettings.ts

```diff
@@ -0,0 +1,53 @@
+import { z } from 'zod';
+import { asyncWrapper } from '../../../../utils/asyncWrapper.js';
+import type { DBTeam, UpdateOtlpSettings } from '@nangohq/types';
+import { requireEmptyQuery, zodErrorToHTTP, isEnterprise } from '@nangohq/utils';
+import { environmentService, featureFlags } from '@nangohq/shared';
+
+const bodyValidation = z
+    .object({
+        endpoint: z.string(),
+        headers: z.record(z.string())
+    })
+    .strict();
+
+export const postSettings = asyncWrapper<UpdateOtlpSettings>(async (req, res) => {
+    const emptyQuery = requireEmptyQuery(req, { withEnv: true });
+    if (emptyQuery) {
+        res.status(400).send({ error: { code: 'invalid_query_params', errors: zodErrorToHTTP(emptyQuery.error) } });
+        return;
+    }
+
+    const val = bodyValidation.safeParse(req.body);
+
+    if (!val.success) {
+        res.status(400).send({
+            error: { code: 'invalid_body', errors: zodErrorToHTTP(val.error) }
+        });
+        return;
+    }
+
+    const { environment, account } = res.locals;
+
+    const isEnabled = await isOtlpEnabled({ account });
+    if (!isEnabled) {
+        res.status(403).send({ error: { code: 'forbidden', message: 'OpenTelemetry export is not enabled for this account' } });
+        return;
+    }
+
+    const { data: settings } = val;
+
+    settings.endpoint = settings.endpoint.trim().replace(/\/$/, '');
+
+    const newSettings = settings.endpoint.length > 0 ? settings : null;
+    await environmentService.editOtlpSettings(environment.id, newSettings);
+
+    res.send(settings);
+});
+
+const isOtlpEnabled = async ({ account }: { account: DBTeam }): Promise<boolean> => {
+    if (isEnterprise) {
+        return true;
+    }
+    return featureFlags.isEnabled('feature:otlp:account', account.uuid, false);
```

Comment:
what about self hosted?

Reply:
it is supposed to be a scale plan only feature so they can only be enterprise or cloud. Am I missing something?

Reply:
it shouldn't be shown in the UI then

Reply:
Agreed. Just check with @bastienbeurier though and we decided to keep it as is for now (aka: failing unless enterprise or cloud with flag enabled) and improve it when making the design of this page better